sdk/keyvault/Azure.Security.KeyVault.Secrets/tests/ChallengeBasedAuthenticationPolicyTests.cs

```diff
@@ -199,6 +199,150 @@ public async Task ReauthenticatesWhenTenantChanged()
             Assert.AreEqual("secret-value", response.Value.Value);
         }
 
+        [Test]
```

Comment:
All clients need the same behavior. Can we test the policy directly in `Azure.Security.KeyVault.Shared`?

Reply:
I'm checking if that's possible.
I wonder if there's a reason why the ChallengeBasedAuthPolicy tests were written inside the Secrets project originally instead of the shared one.  I also noticed that the `Azure.Security.KeyVault.Shared.Test` seems to not have tests defined in it.

Reply:
After further investigation, I found that the shared project should not reference other projects. 

In theory, since the CAE logic introduced in this PR is shared, it can be tested on one of the clients and should work consistently across all clients.

I noticed that `Azure.Security.KeyVault.Administration` already includes tests for `ChallengeBasedAuthenticationPolicy`, so I will add some CAE tests there too.

If we really need to explicitly test CAE for all clients, we could duplicate these tests across the different projects, modifying only the API calls (`GetSecret()`, `GetKey()`, etc.). However, I personally don't see much value with this approach, and it would introduce too much repetitive code.

Reply:
you could probably accomplish this without duplicate code with a base class test in the shared test project and then have a test class in each client test class that inherits from that base class. I did something like this in [identity](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/tests/CredentialTestBase.cs) and we did this in [Core](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/tests/TransportFunctionalTests.cs)